### PR TITLE
Fix the issue of using only inputs_embeds in convbert model

### LINKED ISSUE #21395: UnboundLocalError: local variable 'seq_length' referenced before assignment
### System Info

Hi, I am using the `ConvBertForTokenClassification` model in models.convbert and encountered the bug when passing only `input_embeds` to `forward()`.
The traceback says that at line 833 in modeling_convbert.py
```
if token_type_ids is None:
    if hasattr(self.embeddings, "token_type_ids"):
        buffered_token_type_ids = self.embeddings.token_type_ids[:, :seq_length]
```
The seq_length is unassigned.

I noticed just above this piece of code in
```
elif input_ids is not None:
    input_shape = input_ids.size()
    batch_size, seq_length = input_shape
elif inputs_embeds is not None:
    input_shape = inputs_embeds.size()[:-1]
```
seq_length is not assigned if the program enters `elif inputs_embeds is not None`.

Not sure if it is the `batch_size, seq_length = input_shape` missing for `inputs_embeds` or I am not using the model correctly?

### Who can help?

text models: @ArthurZucker and @younesbelkada

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

passing only `inputs_embeds` and `attention_mask` to `ConvBertForTokenClassification` model.

### Expected behavior

There should be no error.

### PR DESCRIPTION
**Pull request by raghavanone**
Fixes #21395 

@sgugger

### CODE REVIEW

File: tests/models/convbert/test_modeling_convbert.py [hunk 1]
```diff
@@ -437,6 +437,17 @@ def test_torchscript_device_change(self):
...
+    def test_model_for_input_embeds(self):
+        batch_size = 2
+        seq_length = 10
+        inputs_embeds = torch.rand([batch_size, seq_length, 768])
+        config = self.model_tester.get_config()
+        model = ConvBertModel(config=config)
+        model.to(torch_device)
+        model.eval()
+        result = model(inputs_embeds=inputs_embeds)
+        self.assertEqual(result.last_hidden_state.shape, (batch_size, seq_length, config.hidden_size))
```
Comment (ArthurZucker) [inline tests/models/convbert/test_modeling_convbert.py:449]: Nice thanks for thinking of adding a tests. Let's add slow to make sure this is ran on a GPU

**Review by ArthurZucker — APPROVED** [review]
LGTM thanks for the fix 😉 Can you just be a bit more specific on the fix (for futur references) ! 🙏🏻

File: tests/models/convbert/test_modeling_convbert.py (hunk 1 above)
Comment (raghavanone) [reply tests/models/convbert/test_modeling_convbert.py:450]: Done

File: src/transformers/models/convbert/modeling_convbert.py [hunk 2]
```diff
@@ -818,12 +818,13 @@ def forward(
             raise ValueError("You cannot specify both input_ids and inputs_embeds at the same time")
         elif input_ids is not None:
             input_shape = input_ids.size()
-            batch_size, seq_length = input_shape
         elif inputs_embeds is not None:
             input_shape = inputs_embeds.size()[:-1]
+
```
Comment (sgugger) [inline src/transformers/models/convbert/modeling_convbert.py:823]: No need for a new line here.

File: tests/models/convbert/test_modeling_convbert.py [hunk 3]
```diff
@@ -437,6 +437,18 @@ def test_torchscript_device_change(self):
                 loaded = torch.jit.load(os.path.join(tmp, "traced_model.pt"), map_location=torch_device)
                 loaded(inputs_dict["input_ids"].to(torch_device), inputs_dict["attention_mask"].to(torch_device))
 
+    @slow
```
Comment (sgugger) [inline tests/models/convbert/test_modeling_convbert.py:440]:
```suggestion
```

File: tests/models/convbert/test_modeling_convbert.py (hunk 1 above)
Comment (sgugger) [reply tests/models/convbert/test_modeling_convbert.py:450]: All tests are run on GPU, you don't need to put `@slow` for that @ArthurZucker ;-)

**Review by sgugger — COMMENTED** [review]
Thanks for fixing the issue and adding a new test! Can you just edit the title of the PR to be more explicit: it will eventually be in the release notes and no one reading them will understand Fix 21395 ;-)

**Review by sgugger — APPROVED** [review]

File: tests/models/convbert/test_modeling_convbert.py (hunk 1 above)
Comment (ArthurZucker) [reply tests/models/convbert/test_modeling_convbert.py:449]: Oups, I though I had removed that comment! 😓 The device is automatically deduced even if there are no GPU

**Pull request merged**